### ISSUE TITLE
resume training from checkpoint

### ISSUE THREAD

**Issue by NaomiEX** [exiawsh/StreamPETR#151]
Hi, I was wondering whether it is possible to resume training from a particular checkpoint and if so how I would be able to do so

**Comment by NaomiEX** [comment]
Hi, I tried this and the number of iterations is correctly recovered but the learning rate is wrong (I'm training on an RTX 4090)
initial training:
![image](https://github.com/exiawsh/StreamPETR/assets/54797931/e952244c-2fcd-49eb-a9e2-eeec8d66ef0c)

resume:
![image](https://github.com/exiawsh/StreamPETR/assets/54797931/02670ff5-8c1c-40ac-9320-e7a5013411a7)

**Comment by exiawsh** [comment]
Why is the number of iterations different between your two training sessions? Do you want to extend your training time after the first session? Then you need to adjust your learning rate.

**Comment by NaomiEX** [comment]
Yes, I initially only trained for 48 epochs to test it and want to continue the training, I have adjusted the learning rate to match the learning rate of the last epoch:
![image](https://github.com/exiawsh/StreamPETR/assets/54797931/d4040b1d-260b-4728-baca-52da777204da)
But when I try to resume from it, the learning rate is still wrong:
![image](https://github.com/exiawsh/StreamPETR/assets/54797931/d8e9c9a8-8059-4f30-8426-d9ec123ee73b)

**Comment by exiawsh** [comment]
@NaomiEX In fact, it is not recommended to continue training after a complete training cycle. I guess it may be that your batchsize is too small, so the model convergence is not good enough, if you can not reproduce our results.

**Comment by exiawsh** [comment]
@NaomiEX @NaomiEX If you are using a single 4090, I recommend that you set the batchsize to 8 and the learning rate to 2e-4. The training time is about 2 days if you use Res50 as the backbone.

**Comment by NaomiEX** [comment]
At the time, there were other processes using the GPU so I had to use a batch size of 2, I will retrain it now with the recommended settings thanks a lot!
